src/modules/fw_att_control/ecl_pitch_controller.cpp

```diff
@@ -102,11 +102,61 @@ float ECL_PitchController::control_bodyrate(const float dt, const ECL_ControlDat
 		_integrator = math::constrain(_integrator + id * _k_i, -_integrator_max, _integrator_max);
 	}
 
+	// Calculate the contribution of the rate gyro to the actuator demand and high pass filter
+	const float p_term = _rate_error * _k_p * ctl_data.scaler * ctl_data.scaler;
+	float filter_input;
+	if (_k_ff > 0.0f) {
+		// If feed forward is being used then we can monitor the rate error term
+		// because the feed forward should be doing most of the work
+		filter_input = p_term;
+	} else {
+		// Some users will use a zero feed forward gain and rely on the rate error term
+		// to do all the work, resuling in large rate error tranisents. In this scenario
+		// it is better to monitor the gyro feedback contribution only
+		filter_input = ctl_data.body_y_rate * _k_p * ctl_data.scaler * ctl_data.scaler;
+	}
+	const float filt_tconst = _tc / 2.0f;
+	const float filt_coef = fminf(dt, filt_tconst) / filt_tconst;
+	_gyro_contribution_hpf = (1.0f - filt_coef) * _gyro_contribution_hpf + filt_coef * filter_input;
```

Comment:
Since `_gyro_contribution_hpf` actually is the low pass filtered gyro contribution would `_gyro_contribution_lpf` be a better name?

Reply:
fixed